datasets/us/doh_debarred/crawler.py

```diff
@@ -0,0 +1,64 @@
+from zavod import Context
+import csv
+from zavod import helpers as h
+from typing import Dict, Any
+
+FORMATS = ["%Y-%m-%d","%Y%m%d","%y-%m-%d"]
+
+
+def get_address(row: Dict[str, Any]):
+    address = row.get("ADDRESS")
+    if address:
+        address_data = address
+    else:
+        address_data = row.get("ZIP")  # Use ZIP code as ID for empty address
+    return address_data
+
+def crawl_item(context: Context, row: Dict[str, Any]):
+    """Process data from CSV file."""
+    if row["LASTNAME"].strip() or row["FIRSTNAME"].strip():
+        entity = context.make("Person")
+        firstname = row["FIRSTNAME"]
+        midname = row["MIDNAME"]
+        lastname = row["LASTNAME"]
+        name = f"{firstname} {midname} {lastname}".strip()
+        address_data = get_address(row)
+        entity.id = context.make_slug(name, address_data)
+        entity.add("birthDate", h.parse_date(row.get("DOB"), FORMATS))
+        entity.add("name", name)
+    else:
+        entity = context.make("Company")
+        name = row.get("BUSNAME")
+        address_data = get_address(row)
+        entity.id = context.make_slug(name, address_data)
+        entity.add("name", name)
+    entity.add("notes", row.get("GENERAL",None))
+    entity.add("address", row.get("ADDRESS", None))
+    entity.add("topics", "debarment")
+    entity.add("country", "us")
+    sanction = h.make_sanction(context, entity)
+    sanction.add("startDate", h.parse_date(row.get("EXCLDATE"), FORMATS))
+    sanction.add("reason",row.get("EXCLTYPE"))
+    sanction.add("program","US HHS OIG List of Excluded Individuals/Entities ")
+    context.emit(sanction)
+    context.emit(entity, target=True)
+    context.audit_data(row,
```

Comment:
when I crawl the dataset, I get lots of warnings like this:

```
2024-03-14 11:03:32 [warning  ] Unexpected data found          [us_doh_debarred] data={'LASTNAME': 'ARTHUR', 'FIRSTNAME': 'WALLACE', 'MIDNAME': 'R', 'GENERAL': 'PHYSICIAN (MD, DO)', 'DOB': '19540813', 'ADDRESS': '2495 S KRAMERIA STREET', 'ZIP': '80222', 'EXCLTYPE': '1128b4', 'EXCLDATE': '20170119'} dataset=us_doh_debarred
```

we usually use `row.pop` rather than `row.get` or `row[...]` so that each field is removed from the dict as it's used. The idea is that we then "audit" the dict, expecting it to be empty. We normally only merge crawlers when they have no warnings, or if the warnings would be solved after merging, e.g. by adding some position definitions to our PEP position database, or if it's a temporary network issue.

So if we get the dict to be empty (or ignore unused columns as you have) we can merge the PR without warnings. If they later add a column to the CSV, we're automatically alerted because suddenly warnings would show up here https://www.opensanctions.org/issues/, which we check daily. We can then decide if we want to use or ignore the new data.

another benefit of `pop` over `get` is that pop will raise an exception if they key doesn't exist, alerting us to missing data, perhaps from a column header change.

Reply:
Please make sure there aren't any more warnings because all the columns that aren't used are ignored. Please ask if there are still warnings you're not sure of.